### PR TITLE
Updated Automatic Speech Recognition using CTC example for Keras v3

### PR DESCRIPTION
**Pull request by lpizzinidev**
Updates the "Automatic Speech Recognition using CTC" example to support Keras v3.

### CODE REVIEW

File: examples/audio/ctc_asr.py [hunk 1]
```diff
@@ -244,16 +249,74 @@ def encode_single_sample(wav_file, label):
 """
 
 
+# Reference: https://github.com/keras-team/keras/blob/ec67b760ba25e1ccc392d288f7d8c6e9e153eea2/keras/legacy/backend.py#L674-L711
+def ctc_label_dense_to_sparse(labels, label_lengths):
```
Comment (fchollet) [inline examples/audio/ctc_asr.py:253]: Rather than rewriting this code, you can just use the built-in Keras 3 loss function `keras.losses.CTC`. I expect it will also enable the code example to run with all backends.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/audio/ctc_asr.py (hunk 1 above)
Comment (lpizzinidev) [reply examples/audio/ctc_asr.py]: Thanks for the feedback 👍
After removing the legacy code we still have some references to `tf` in the example and I'm not sure this can be made backend-agnostic.
Please let me know if I should substitute the remaining `tf` references.

File: examples/audio/ctc_asr.py [hunk 2]
```diff
@@ -320,7 +307,7 @@ def build_model(input_dim, output_dim, rnn_layers=5, rnn_units=128):
     # Optimizer
     opt = keras.optimizers.Adam(learning_rate=1e-4)
     # Compile the model and return
-    model.compile(optimizer=opt, loss=CTCLoss)
+    model.compile(optimizer=opt, loss=keras.losses.ctc)
```
Comment (fchollet) [inline examples/audio/ctc_asr.py:310]: Prefer using `CTC()` (ends up running the same thing but it's more idiomatic)

File: examples/audio/ctc_asr.py [hunk 3]
```diff
@@ -337,11 +324,38 @@ def build_model(input_dim, output_dim, rnn_layers=5, rnn_units=128):
...
+
+
+def ctc_decode(y_pred, input_length, greedy=True, beam_width=100, top_paths=1):
+    input_shape = tf.shape(y_pred)
+    num_samples, num_steps = input_shape[0], input_shape[1]
+    y_pred = tf.math.log(tf.transpose(y_pred, perm=[1, 0, 2]) + keras.backend.epsilon())
+    input_length = tf.cast(input_length, tf.int32)
+
+    if greedy:
+        (decoded, log_prob) = tf.nn.ctc_greedy_decoder(
```
Comment (fchollet) [inline examples/audio/ctc_asr.py:337]: So, we're going to have to use TF for this and `ctc_beam_search_decoder` I guess, unless we implement them as new backend ops.

**Review by fchollet — COMMENTED** [review]
LGTM, thank you! You can add the generated files.

File: examples/audio/ctc_asr.py (hunk 3 above)
Comment (lpizzinidev) [reply examples/audio/ctc_asr.py:337]: Again, thanks for the feedback 👍
I created an [issue](https://github.com/keras-team/keras/issues/19227) to address this.
Please let me know if I should change the description or add/remove details.
Thanks!
